vaadinfrontend/src/main/java/life/qbic/datamanager/views/projectOverview/ProjectOverviewHandler.java

```diff
@@ -0,0 +1,71 @@
+package life.qbic.datamanager.views.projectOverview;
+
+import com.vaadin.flow.component.Text;
+import com.vaadin.flow.data.renderer.ComponentRenderer;
+import life.qbic.projectmanagement.application.finances.offer.OfferLookupService;
+import life.qbic.projectmanagement.domain.finances.offer.OfferPreview;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Objects;
+
+/**
+ * <b>Handler</b>
+ *
+ * <p>Orchestrates the layout {@link ProjectOverviewLayout} and determines how the components behave.</p>
+ *
+ * @since <version tag>
+ */
+@Component
+public class ProjectOverviewHandler implements ProjectOverviewHandlerInterface{
+
+    private ProjectOverviewLayout registeredProjectOverview;
+    private final OfferLookupService offerLookupService;
+
+    ProjectOverviewHandler(@Autowired OfferLookupService offerLookupService){
+        Objects.requireNonNull(offerLookupService);
+        this.offerLookupService = offerLookupService;
+    }
+
+    @Override
+    public void handle(ProjectOverviewLayout layout) {
+        if (registeredProjectOverview != layout) {
+            this.registeredProjectOverview = layout;
+            addClickListeners();
+        }
+    }
+
+    private void addClickListeners() {
+        registeredProjectOverview.searchDialog.cancel.addClickListener(e -> registeredProjectOverview.searchDialog.close());
+
+        registeredProjectOverview.create.addClickListener( e-> {
+            loadItemsWithService(offerLookupService);
+            registeredProjectOverview.searchDialog.open();
+        });
+
+        registeredProjectOverview.searchDialog.ok.addClickListener(e -> {
+            //check if value is selected
+            if(registeredProjectOverview.searchDialog.searchField.getOptionalValue().isPresent()){
+                forwardSelectedOffer();
+            }
+        });
+    }
+
+    private void forwardSelectedOffer() {
+        //todo forward to service to load into create offer UI
+        OfferPreview selectedOfferPreview = registeredProjectOverview.searchDialog.searchField.getValue();
+        registeredProjectOverview.add(new Text(selectedOfferPreview.offerId().id()+", "+selectedOfferPreview.getProjectTitle().title()));
+        registeredProjectOverview.searchDialog.close();
+    }
```

Comment:
For inter-page navigation we need a strategy. How do we pass information from on page to another? or from one layout to another?